### ISSUE TITLE
[Bug]: think重复出现感叹号

### ISSUE THREAD

**Issue by ld555** [1CatAI/1Cat-vLLM-0.0.1#30]
### Your current environment

{
    "id": "chatcmpl-af6e1368bfc5cdb7",
    "object": "chat.completion",
    "created": 1778213292,
    "model": "qwen3.6-27b-awq",
    "choices": [
        {
            "index": 0,
            "message": {
                "role": "assistant",
                "content": null,
                "refusal": null,
                "annotations": null,
                "audio": null,
                "function_call": null,
                "tool_calls": [],
                "reasoning": "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
            },
            "logprobs": null,
            "finish_reason": "length",
            "stop_reason": null,
            "token_ids": null
        }
    ],
    "service_tier": null,
    "system_fingerprint": null,
    "usage": {
        "prompt_tokens": 13,
        "total_tokens": 525,
        "completion_tokens": 512,
        "prompt_tokens_details": null
    },
    "prompt_logprobs": null,
    "prompt_token_ids": null,
    "kv_transfer_params": null
}

### 🐛 Describe the bug

{
    "id": "chatcmpl-af6e1368bfc5cdb7",
    "object": "chat.completion",
    "created": 1778213292,
    "model": "qwen3.6-27b-awq",
    "choices": [
        {
            "index": 0,
            "message": {
                "role": "assistant",
                "content": null,
                "refusal": null,
                "annotations": null,
                "audio": null,
                "function_call": null,
                "tool_calls": [],
                "reasoning": "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
            },
            "logprobs": null,
            "finish_reason": "length",
            "stop_reason": null,
            "token_ids": null
        }
    ],
    "service_tier": null,
    "system_fingerprint": null,
    "usage": {
        "prompt_tokens": 13,
        "total_tokens": 525,
        "completion_tokens": 512,
        "prompt_tokens_details": null
    },
    "prompt_logprobs": null,
    "prompt_token_ids": null,
    "kv_transfer_params": null
}

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by ld555** [comment]
docker run -d \
  --name qwen36-27b-awq \
  --gpus '"device=0,1"' \
  --ipc=host \
  --ulimit memlock=-1 \
  --ulimit stack=67108864 \
  -p 8000:8000 \
  -v /data/modelscope:/models:ro \
  -e VLLM_ATTENTION_BACKEND=FLASH_ATTN_V100 \
  -e VLLM_SM70_ENABLE_LM_HEAD_FASTPATH=1 \
  1cat-vllm-sm70:0.0.3 \
  --model /models/Qwen3.6-27B-AWQ \
  --served-model-name qwen3.6-27b-awq \
  --trust-remote-code \
  --quantization awq \
  --dtype float16 \
  --tensor-parallel-size 2 \
  --gpu-memory-utilization 0.90 \
  --max-model-len 131072 \
  --max-num-seqs 1 \
  --max-num-batched-tokens 4096 \
  --skip-mm-profiling \
  --reasoning-parser qwen3 \
  --default-chat-template-kwargs '{"enable_thinking": true}' \
  --compilation-config '{"cudagraph_mode":"full_and_piecewise","cudagraph_capture_sizes":[1,2]}' \
  --enable-auto-tool-choice \
  --tool-call-parser qwen3_coder \
  --host 0.0.0.0 \
  --port 8000

**Comment by bayley** [comment]
Should be the same issue as #24 , #26 , #29 : the implementation of FLASH_ATTN_V100 here is buggy, sometimes producing numeric errors in the output. Currently fixed in the downstream https://github.com/humanjesse/vllm-v100, but I think the actual issue is there is a lot of CUDA code being generated by Claude and edge cases are being missed.

**Comment by yangzhuxinyzx** [comment]
这个和 #24 / #29 属于同一类旧版 `FLASH_ATTN_V100` 输出质量问题，发生在 0.0.3 时代的 Flash-V100 decode/paged KV 路线。

请先升级到 1Cat-vLLM 1.2.0：

https://github.com/1CatAI/1Cat-vLLM/releases/tag/v1.2.0

1.2.0 已经重新接入 Flash-V100 backend，并更新了 Qwen3.6-27B-AWQ 推荐启动命令。建议先用 README public command 复测，不要额外启用旧实验 env。

如果 1.2.0 仍然出现 `reasoning` 里大量感叹号，请贴新的：

- wheel/version 信息
- 完整启动命令
- 请求 body
- 完整返回 JSON

这样我们可以按当前 1.2.0 路线继续定位。

**Comment by yangzhuxinyzx** [comment]
The repeated `!` tokens indicate generation-token corruption rather than a parser-only presentation issue. In the reported shape, current eligible decode is the D256/GQA 6:1 Flash-V100 XQA paged route, but the historical report has no current route or logit trace.

No source change is being made without a current V100 reproduction. The required localization matrix is: Triton, Flash+XQA, Flash scalar decode, and Triton prefill+Flash decode under fixed sampling/CUDA Graph state, with first divergent layer and logits top-k recorded. The issue remains open pending that evidence.
